lib/segment/src/index/field_index/numeric_index/mmap_numeric_index.rs

```diff
@@ -0,0 +1,310 @@
+            histogram,
+            indexed_points_count: config.indexed_points_count,
+            max_values_per_point: config.max_values_per_point,
+            point_to_values,
+        })
+    }
+
+    pub fn clear(self) -> OperationResult<()> {
+        let files = self.files();
+        drop(self);
+        for file in files {
+            std::fs::remove_file(file)?;
+        }
+        Ok(())
+    }
+
+    pub fn files(&self) -> Vec<PathBuf> {
+        let mut files = vec![
+            self.path.join(PAIRS_PATH),
+            self.path.join(DELETED_PATH),
+            self.path.join(CONFIG_PATH),
+        ];
+        files.extend(self.point_to_values.files());
+        files.extend(Histogram::<T>::files(&self.path));
+        files
+    }
+
+    pub fn flusher(&self) -> Flusher {
+        self.deleted.flusher()
+    }
+
+    pub(super) fn check_values_any(
+        &self,
+        idx: PointOffsetType,
+        check_fn: impl Fn(&T) -> bool,
+    ) -> bool {
+        if self.deleted.get(idx as usize).as_deref() == Some(&false) {
+            self.point_to_values
+                .check_values_any(idx, |v| check_fn(T::from_referenced(&v)))
+        } else {
+            false
+        }
+    }
+
+    pub fn get_values(&self, idx: PointOffsetType) -> Option<Box<dyn Iterator<Item = T> + '_>> {
+        if self.deleted.get(idx as usize).as_deref() == Some(&false) {
+            Some(Box::new(
+                self.point_to_values
+                    .get_values(idx)?
+                    .map(|v| *T::from_referenced(&v)),
+            ))
+        } else {
+            None
+        }
+    }
+
+    pub fn values_count(&self, idx: PointOffsetType) -> Option<usize> {
+        if self.deleted.get(idx as usize).as_deref() == Some(&false) {
+            self.point_to_values.get_values_count(idx)
+        } else {
+            None
+        }
+    }
+
+    pub(super) fn total_unique_values_count(&self) -> usize {
+        self.pairs.len()
+    }
+
+    pub(super) fn values_range(
+        &self,
+        start_bound: Bound<Point<T>>,
+        end_bound: Bound<Point<T>>,
+    ) -> impl Iterator<Item = PointOffsetType> + '_ {
+        self.values_range_iterator(start_bound, end_bound)
+            .map(|Point { idx, .. }| idx)
+    }
+
+    pub(super) fn orderable_values_range(
+        &self,
+        start_bound: Bound<Point<T>>,
+        end_bound: Bound<Point<T>>,
+    ) -> impl DoubleEndedIterator<Item = (T, PointOffsetType)> + '_ {
+        self.values_range_iterator(start_bound, end_bound)
+            .map(|Point { val, idx }| (val, idx))
+    }
+
+    pub(super) fn remove_point(&mut self, idx: PointOffsetType) {
+        if let Some(mut deleted) = self.deleted.get_mut(idx as usize) {
+            *deleted = true;
+        }
+    }
+
+    pub(super) fn get_histogram(&self) -> &Histogram<T> {
+        &self.histogram
+    }
+
+    pub(super) fn get_points_count(&self) -> usize {
+        self.indexed_points_count
```

Comment:
Let's account for deleted here (it will be displayed to user, so we need as exact as possible)

Reply:
done